src/app/kbv/fields.js

```diff
@@ -1,4 +1,24 @@
-const { numericWithOptionalDecimalZeros } = require("./fieldsHelper");
+const {
+  numericWithOptionalDecimalZeros,
+  numericWithOptionalDecimal,
+  numericWithRequiredDecimals,
+} = require("./fieldsHelper");
+
+const amountFieldValidateRequiredAndAmountWithDecimals = {
+  type: "text",
+  validate: [
+    "required",
+    {
+      type: "regexNumeric",
+      fn: (value) => !!value.match(numericWithOptionalDecimal),
+    },
+    {
+      type: "regex",
```

Comment:
Could we name these a bit clearer to distinguish them - maybe `regexNumeric` and `regexPoundsAndPence`?